### PR TITLE
Discuss: Remove public dir support

### PR DESCRIPTION
**Pull request by ruflin**
The public dir was needed in the past to build packages. But with the recent changes to the registry this has become obsolete. Instead the registry can be directly pointed at a directory with packages. In the package storage this directory is called `packages`, I suggest we use this as a default. The registry itself could have its packages in the same directory.

This PR is not complete, it is more to discuss the change.

### CODE REVIEW

File: handler.go [hunk 1]
```diff
@@ -39,6 +40,7 @@ func catchAll(public http.FileSystem, cacheTime time.Duration) func(w http.Respo
 func determineResourcePath(r *http.Request, public http.FileSystem) (string, error) {
 	path := r.URL.Path
 
+	path = strings.Replace(path, "/package", "", 1)
```
Comment (ruflin) [inline handler.go:43]: This change is odd but is needed as now all is at the top level

**Comment by ruflin** [comment]
@mtojek I assume this also ties into your change you have planned for multiple paths?

**Comment by mtojek** [comment]
Yes, it's draft currently: https://github.com/elastic/package-registry/pull/525

**Comment by mtojek** [comment]
I planned to leave `PublicDir` for legacy purposes:

```golang
type Config struct {
	PublicDir           string        `config:"public_dir"` // left for legacy purposes
	PackagesPaths       []string      `config:"packages_paths"`
	CacheTimeSearch     time.Duration `config:"cache_time.search"`
	CacheTimeCategories time.Duration `config:"cache_time.categories"`
	CacheTimeCatchAll   time.Duration `config:"cache_time.catch_all"`
}
```

**Comment by ruflin** [comment]
Directly closing this as https://github.com/elastic/package-registry/pull/525 already covers the idea from this PR .

@mtojek Lets try to get rid of all legacy before 7.9, we will build up enough over the next months :-)

**Pull request closed without merging**